pyanaconda/format_dasd.py

```diff
@@ -0,0 +1,195 @@
+
+    def _is_ldl_dasd(self, disk):
+        """Is it an LDL DASD?"""
+        return self._is_dasd(disk) and blockdev.s390.dasd_is_ldl(disk.name)
+
+    def _get_unformatted_dasds(self, disks):
+        """Returns a list of unformatted DASDs."""
+        result = []
+
+        if not self._can_format_unformatted:
+            log.debug("We are not allowed to format unformatted DASDs.")
+            return result
+
+        for disk in disks:
+            if self._is_unformatted_dasd(disk):
+                log.debug("Found unformatted DASD: %s", self.get_dasd_info(disk))
+                result.append(disk)
+
+        return result
+
+    def _get_ldl_dasds(self, disks):
+        """Returns a list of LDL DASDs."""
+        result = []
+
+        if not self._can_format_ldl:
+            log.debug("We are not allowed to format LDL DASDs.")
+            return result
+
+        for disk in disks:
+            if self._is_ldl_dasd(disk):
+                log.debug("Found LDL DASD: %s", self.get_dasd_info(disk))
+                result.append(disk)
+
+        return result
+
+    def update_restrictions(self, data):
+        """Read kickstart data to update the restrictions."""
+        self._can_format_unformatted = data.zerombr.zerombr
+        self._can_format_ldl = data.clearpart.cdl
+
+    def search_disks(self, disks):
+        """Search for a list of disks for DASDs to format."""
+        self._dasds = list(set(self._get_unformatted_dasds(disks) + self._get_ldl_dasds(disks)))
+
+    def should_run(self):
+        """Should we run the formatting?"""
+        return bool(self._dasds)
+
+    def do_format(self, disk):
+        """Format a disk."""
+        try:
+            self.report.emit(_("Formatting %s") % self.get_dasd_info(disk))
+            blockdev.s390.dasd_format(disk.name)
+        except blockdev.S390Error as err:
+            self.report.emit(_("Failed formatting %s") % self.get_dasd_info(disk))
+            log.error(err)
+
+    def run(self, storage, data):
+        """Format all found DASDs and update the storage."""
+        # Check if we have something to format.
+        if not self._dasds:
+            self.report.emit(_("Nothing to format"))
+            return
+
+        # Format all found DASDs.
+        self.report.emit(_("Formatting DASDs"))
+        for disk in self._dasds:
+            self.do_format(disk)
+
+        # Update the storage.
+        self.report.emit(_("Probing storage"))
+        storage_initialize(storage, data, storage.devicetree.protected_dev_names)
+
+        # Update also the storage snapshot to reflect the changes.
+        if on_disk_storage.created:
+            on_disk_storage.dispose_snapshot()
+        on_disk_storage.create_snapshot(storage)
+
+    @staticmethod
+    def run_automatically(storage, data, callback=None):
+        """Run the DASD formatting automatically."""
+
+        # Is it an automated install?
```

Comment:
Please remove these comments it makes code longer not clearer.

Reply:
Fixed.